src/Modules/UngradedModules/FreeModuleHom.jl

```diff
@@ -74,7 +74,10 @@ function matrix(f::FreeModuleHom)
   return f.matrix
 end
 
-(h::FreeModuleHom)(a::AbstractFreeModElem) = image(h, a)
+function (h::FreeModuleHom)(a::AbstractFreeModElem)
+  @assert parent(a) === domain(h)
```

Comment:
```suggestion
  @req parent(a) === domain(h) "invalid input"
```
we usually use `@req` for input sanitation